src/classes/RD2_EntryFormController.cls

```diff
@@ -150,6 +150,24 @@ public with sharing class RD2_EntryFormController {
         return isMissingRequiredPermissions;
     }
 
+    /**
+     * @description Validate that the current contact has a US address
+     * @return True if Contact address is a US address
+     */
+    @AuraEnabled
+    public static Boolean contactHasUSAddress(Id recordId) {
+        Contact rdContact = [SELECT MailingState, MailingCountry FROM Contact WHERE Id = :recordId];
+        if (String.isNotBlank(rdContact.MailingState)
+            && (rdContact.MailingCountry == 'US'
```

Comment:
To be on the safe side, I think this should also check for:
- U.S.
- U.S.A.
- United States of America

Reply:
That is correct, it is re-written to use `ADDR_Validator_UTIL.isUSCountry(country);`. However, I am checking the code now if we should validate the country in the JS instead of Apex...

Reply:
Just for performance reasons, I think it should stay in JS. Your point (below) about supporting other countries down the road is a good one. Maybe this could use a constant such `ELEVATE_SUPPORTED_COUNTRIES` that is an array of just US countries for now. 

I see an opportunity for shared content between this widget and GE if we use a utility component, but at the same time I'm not sure how GE is handling address/currency verification to display the CC widget.

Reply:
@ptewson-sfdo , @psadie-sfdo : do you validate the currency and the Contact address for a GE purchase?

Reply:
I don't believe we do. As of right now we set the currency code for every purchase transaction using `UserInfo.getDefaultCurrency()`.

Reply:
@force2b , the change has been made to validate the Contact address. We check Mailing Country only.
The Contact is validated regardless if the "Donor Type" is Account or Contact. The AC5 was not specific about the "Donor Type" so the implementation is not specific either.

CC @mpusto , @rponti-sforg